metacat-common-server/src/main/java/com/netflix/metacat/common/server/connectors/ConnectorFactoryDecorator.java

```diff
@@ -32,24 +36,33 @@ public ConnectorFactoryDecorator(@NonNull final ConnectorPlugin connectorPlugin,
         this.connectorContext = connectorContext;
 
         this.rateLimiter = connectorContext.getApplicationContext().getBean(RateLimiter.class);
+        this.authorization = connectorContext.getApplicationContext().getBean(Authorization.class);
 
-        // we can cache this config at startup since this is the connector level config
-        // that does not change later. Actual decision to enable and enforce throttling
-        // is in the rate limiter implementation which is more dynamic and accommodates
-        // changes from the Metacat dynamic configuration.
+        // we can cache these configs at startup since this is the connector level config
+        // that does not change later. Actual decision to enable and enforce throttling and/or
+        // authorization is in the rate limiter or auth implementations which are more dynamic
+        // and accommodate changes from the Metacat dynamic configuration.
         this.rateLimiterEnabled = isRateLimiterEnabled();
+        this.authorizationEnabled = isAuthorizationEnabled();
+        this.validationEnabled = isValidationEnabled();
     }
 
     @Override
     public ConnectorCatalogService getCatalogService() {
         ConnectorCatalogService service = delegate.getCatalogService();
 
-        if (rateLimiterEnabled) {
-            log.info("Creating rate-limited connector catalog services for connector-type: {}, "
-                         + "plugin-type: {}, catalog: {}, shard: {}",
+        if (validationEnabled) {
+            log.info("Creating validating connector catalog services for connector-type: {}, "
+                    + "plugin-type: {}, catalog: {}, shard: {}",
                 connectorContext.getConnectorType(), connectorPlugin.getType(),
                 connectorContext.getCatalogName(), connectorContext.getCatalogShardName());
-            service = new ThrottlingConnectorCatalogService(service, rateLimiter);
+            service = new ValidatingConnectorCatalogService(
+                service,
+                rateLimiter,
+                rateLimiterEnabled,
+                authorization,
+                authorizationEnabled
```

Comment:
The authorization object should have the capability to determine its own enabled status, rather than having it flagged externally.